tests/lms/views/predicates/_lti_launch_test.py

```diff
@@ -0,0 +1,264 @@
+from unittest import mock
+
+from pyramid.testing import DummyRequest
+import pytest
+
+from lms.models import ModuleItemConfiguration
+from lms.views.predicates import (
+    DBConfigured,
+    CanvasFile,
+    URLConfigured,
+    Configured,
+    AuthorizedToConfigureAssignments,
+)
+
+
+class TestDBConfigured:
+    def test_when_theres_a_matching_assignment_config_in_the_db(self, pyramid_request):
+        pyramid_request.params = {
+            "resource_link_id": "test_resource_link_id",
+            "tool_consumer_instance_guid": "test_tool_consumer_instance_guid",
+        }
+
+        assert (
```

Comment:
There is rather a lot of repetition of this sequence of asserts in the file. Perhaps consider extracting a helper function, eg.

```py
def check_configured(request, predicate_class, predicate_value, expect_match):
   ...
```

Reply:
I usually don't like shared test helper functions like this, except where really necessary, 'cos I think they make tests harder to understand and couple tests together. I'd suffer a certain amount of test code duplication instead. But I've rewritten these tests to reduce the duplication by using `parametrize`